cmd/uhc/cluster/list/list.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/openshift-online/uhc-cli/pkg/config"
+	"github.com/openshift-online/uhc-cli/pkg/util"
+	"github.com/openshift-online/uhc-sdk-go/pkg/client"
+	"github.com/openshift-online/uhc-sdk-go/pkg/client/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	debug     bool
+	parameter []string
+	managed   bool
+}
+
+var Cmd = &cobra.Command{
+	Use:   "list [MANAGED | UNMANAGED] ",
```

Comment:
I think this description doesn't match the actual usage. If I understand correctly the usage will be one of these:

- `uhc cluster list` - Should list all clusters, not only the managed ones.
- `uhc cluster list --managed` - Should list only the managed clusters.
- `uhc cluster list --managed=true` - Should list only the managed clusters, like the previous one.
- `uhc cluster list --managed=false` - Should list only the non managed clusters.

So the usage text should be `list`, the Cobra library will automatically add a `[flags]` part, so what the user will see is something like this:

```
$ uhc cluster list -h
Usage:
  uhc cluster list [flags]

Flags:
  --debug string ...
```

To implement that you will need to use a string flag, not bool, as I think that boolean flags don't accept `true|false` parameters, you will have to check that explicitly.

Reply:
Ah good catch, I didn't update the description. And I was having a hard time using it as a bool, so I'm happy to change it to a string

Reply:
Cobra looks like it doesn't support a String flag without an argument:
```
uhc cluster list --managed
Error: flag needs an argument: --managed
```
Do you know of a way around this?

Reply:
Looks like recent versions of `pflag` and `cobra` allow `true|false` as optional values for boolean flags. So you an define the flag like this:

```go
flags.BoolVar(
        &args.managed,
        "managed",
        false,
        "Blah, blah, ...",
)
```

You can then use `--managed`, `--managed=true` or `--managed=false`. Using `--managed` is equivalent to `--managed=true`.

In order to check if the parameter was explicitly specified by the user, you can use the `Changed` method of the `FlagSet` type. For example:

```go
searchQuery := ""
if cmd.Flags().Changed("managed") {
        // The parameter was explicitly given by the user, so we need to add
        // the criteria to the search query:
        searchQuery = fmt.Sprintf("managed = '%v'", args.managed)
} else {
        // The parameter wasn't explicitly given by the user, so we don't add
        // it to the search query.
}
```

Reply:
Thanks!